services/121-service/src/program-financial-service-provider-configurations/program-financial-service-provider-configurations.service.ts

```diff
@@ -1,124 +1,427 @@
+          where: {
+            programFinancialServiceProviderConfigurationId: Equal(
+              configIdToCheckForDuplicates,
+            ),
+            name: In(properties.map((property) => property.name)),
+          },
+        });
+      for (const property of exisingProperties) {
+        errors.push(
+          `Property with name ${property.name} already exists for this configuration`,
+        );
+      }
+    }
+
+    if (errors.length > 0) {
+      const errorsString = errors.join(' ');
+
+      throw new HttpException(errorsString, HttpStatus.BAD_REQUEST);
+    }
+  }
+
+  public async updateProperty({
+    programId,
+    programFspConfigurationName,
+    propertyName,
+    property,
+  }: {
+    programId: number;
+    programFspConfigurationName: string;
+    propertyName: FinancialServiceProviderConfigurationProperties;
+    property: UpdateProgramFinancialServiceProviderConfigurationPropertyDto;
+  }): Promise<ProgramFinancialServiceProviderConfigurationPropertyResponseDto> {
+    // Find the configuration
+
+    const config = await this.getProgramFspConfigurationOrThrow(
+      programId,
+      programFspConfigurationName,
+    );
+    const existingProperty =
+      await this.getProgramFspConfigurationPropertyOrThrow(
+        config.id,
+        propertyName,
+      );
+
+    existingProperty.value =
+      ProgramFinancialServiceProviderConfigurationMapper.mapPropertyDtoValueToEntityValue(
+        property.value,
+        existingProperty.name,
+      );
+
+    const savedProperty =
+      await this.programFspConfigurationPropertyRepository.save(
+        existingProperty,
+      );
+
+    return ProgramFinancialServiceProviderConfigurationMapper.mapPropertyEntityToDto(
+      savedProperty,
+    );
+  }
+
+  public async deleteProperty({
+    programId,
+    programFspConfigurationName,
+    propertyName,
+  }: {
+    programId: number;
+    programFspConfigurationName: string;
+    propertyName: FinancialServiceProviderConfigurationProperties;
+  }): Promise<void> {
+    const config = await this.getProgramFspConfigurationOrThrow(
+      programId,
+      programFspConfigurationName,
+    );
+    const existingProperty =
+      await this.getProgramFspConfigurationPropertyOrThrow(
+        config.id,
+        propertyName,
+      );
+    await this.programFspConfigurationPropertyRepository.delete({
+      id: existingProperty.id,
+    });
+  }
+
+  private async createProperties(
+    programFspConfigurationId: number,
+    inputProperties: CreateProgramFinancialServiceProviderConfigurationPropertyDto[],
+  ): Promise<ProgramFinancialServiceProviderConfigurationPropertyEntity[]> {
+    const propertiesToSave =
+      ProgramFinancialServiceProviderConfigurationMapper.mapPropertyDtosToEntities(
+        inputProperties,
+        programFspConfigurationId,
+      );
+    return this.programFspConfigurationPropertyRepository.save(
+      propertiesToSave,
+    );
+  }
+
+  private async overwriteProperties(
```

Comment:
"overwrite" is not really what happens, or, it is ambigue.
Suggest changing to deleteAllAndCreateProperties?

Reply:
I disagree.. how else do you interpret overwrite? I think it's easier to read than deleteAllAndCreateProperties

Reply:
Maybe overwritePropertiesOfConfiguration?

Reply:
> I disagree.. how else do you interpret overwrite?
That they keep the same IDs.

Reply:
> That they keep the same IDs.

Why they are overwritten right?

Reply:
Maybe `overwritePropertyEntities`? So it's clear that it's about the whole entities, not just the values?

Reply:
I still think that "overwrite" is not what technically happens. How about "replaceProperties()"?
If you do not agree with this suggestion, then leave it as overwriteProperties(). LOGAF medium.